mule-packager/src/test/java/org/mule/tools/api/packager/builder/DomainBundlePackageBuilderTest.java

```diff
@@ -10,135 +10,224 @@
-    when(fileMock.exists()).thenReturn(true);
-    builder.withDestinationFile(fileMock);
+    expectedException.expectMessage("The folder must must exists");
+    builder.withMaven(new File("fake"));
   }
 
   @Test
-  public void withMavenNullTest() {
+  public void setNullDomainFolder() {
     expectedException.expect(IllegalArgumentException.class);
     expectedException.expectMessage("The folder must not be null");
-    builder.withMaven(null);
+    builder.withDomain(null);
   }
 
   @Test
-  public void withDomainNullTest() {
+  public void setNonExistentDomainFolder() {
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("The folder must not be null");
-    builder.withDomain(null);
+    expectedException.expectMessage("The folder must must exists");
+    builder.withDomain(new File("fake"));
   }
 
   @Test
-  public void withApplicationsNullTest() {
+  public void setNullApplicationsFolder() {
     expectedException.expect(IllegalArgumentException.class);
     expectedException.expectMessage("The folder must not be null");
     builder.withApplications(null);
   }
 
   @Test
-  public void createPackageTest() throws IOException {
-    DomainBundlePackageBuilder builderSpy = spy(builder);
+  public void setNonExistentApplicationsFolder() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("The folder must must exists");
+    builder.withApplications(new File("fake"));
+  }
 
-    doNothing().when(builderSpy).createDeployableFile();
+  @Test(expected = IllegalArgumentException.class)
+  public void createPackageNullOriginalFolderPath() throws IOException {
+    builder.createPackage(null, destinationFile.toPath());
+  }
 
-    File applicationsFolder = temporaryOriginFolder.newFolder("applications");
-    File domainFolder = temporaryOriginFolder.newFolder("domain");
-    File metaInfFolder = temporaryOriginFolder.newFolder("META-INF");
-    File mavenFolder = new File(metaInfFolder, "maven");
-    mavenFolder.mkdir();
+  @Test(expected = IllegalArgumentException.class)
+  public void createPackageNonExistentOriginalFolderPath() throws IOException {
+    File fileMock = mock(File.class);
+    when(fileMock.exists()).thenReturn(false);
+    builder.createPackage(new File("fake").toPath(), destinationFile.toPath());
+  }
 
-    builderSpy.createPackage(destinationFile, temporaryOriginFolder.getRoot().getAbsolutePath());
+  @Test(expected = IllegalArgumentException.class)
+  public void createPackageNullDestinationPath() throws IOException {
+    builder.createPackage(fakeTargetFolder.getRoot().toPath(), null);
+  }
 
-    verify(builderSpy, times(1)).withDestinationFile(destinationFile);
-    verify(builderSpy, times(1)).withDomain(domainFolder);
-    verify(builderSpy, times(1)).withApplications(applicationsFolder);
-    verify(builderSpy, times(1)).withMaven(mavenFolder);
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
Could you please change "@Test(expected=..." and use expected exception as it is what is used in the whole project? Actually, the expectedException is already used in this test class so either change all of them to the project's default or if there is a good reason change all of them to the annotation and we should create a jira for why we should use this instead of the ExpectedException instance.

Reply:
There is really no strong reason to favor one over another one. 
This I mention you when you started using the expectedException. 
The only compelling to use the expectedException is what you have done when you actually want to validate the message of the exception which doesn't really apply to all scenarios. 

Then again you have a point for this class that has been using the expectedException instance so I'll change it for consistency inside the class I will not say we should change it across the hole project.

Reply:
done